### PR TITLE
SE: Forget field state on await

### PR DESCRIPTION
**Pull request by pavel-mikula-sonarsource**
Related to #5973

### CODE REVIEW

File: analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Operations.cs [hunk 1]
```diff
@@ -713,5 +713,26 @@ public void EventReference_SetsNotNull()
             validator.ValidateTag("AfterAdd", x => x.HasConstraint(ObjectConstraint.NotNull).Should().BeTrue());
             validator.ValidateTag("AfterRemove", x => x.HasConstraint(ObjectConstraint.NotNull).Should().BeTrue());
         }
+
+        [TestMethod]
+        public void Await_ForgetsFieldStates()
+        {
+            const string code = @"
+private object field;
```
Comment (martin-strecker-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Operations.cs:721]: This should be extended (parameterized test?) to the following list
* private object field01;
* private readonly object field02;
* private static object field03;
* private static readonly object field03;
* private volatile object field04;
* private usafe object field05;

Same list with other access modifiers:

* public
* protected
* internal

File: analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Operations.cs [hunk 2]
```diff
@@ -713,5 +713,26 @@ public void EventReference_SetsNotNull()
...
+
+public async System.Threading.Tasks.Task Main(System.Threading.Tasks.Task T)
+{
+    field = null;
+    Tag(""Before"", field);
+    await T;
+    Tag(""After"", field);
+}";
+            var addConstraint = new PostProcessTestCheck(OperationKind.Literal, x => x.SetOperationConstraint(LockConstraint.Held));    // Persisted constraint
```
[the remark this answers is not in the document]
Comment (pavel-mikula-sonarsource) [reply analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Operations.cs:730]: Postponed after rebase (not addressed yet)

File: analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Operations.cs (hunk 1 above)
Comment (pavel-mikula-sonarsource) [reply analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Operations.cs:721]: The scope is the await. We don't need retest your PR again.

File: analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Operations.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (pavel-mikula-sonarsource) [reply analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Operations.cs:784]: Not applicabile anymore 😃

**Review by martin-strecker-sonarsource — APPROVED** [review]
LGTM

**Pull request merged**